projects/rocfft/library/src/plan.cpp

```diff
@@ -1751,29 +1751,35 @@ static std::unique_ptr<ExecPlan> BuildSingleDevicePlan(NodeMetaData&         roo
         // since we are trying enumerating solutions now
         if(TuningBenchmarker::GetSingleton().IsInitializingTuning() == false)
         {
-            // Solutions are tuned on the common case of contiguous
-            // transforms.  They are not necessarily correct for
-            // non-default strides.
-            if(rocfft_plan_t::is_contiguous(
-                   execPlan.rootPlan->length, execPlan.rootPlan->inStride, execPlan.rootPlan->iDist)
-               && rocfft_plan_t::is_contiguous(execPlan.rootPlan->outputLength,
-                                               execPlan.rootPlan->outStride,
-                                               execPlan.rootPlan->oDist))
+            // Solutions do not consider strides.  Even-length real
+            // transforms need special consideration, since the
+            // even-length optimization is only valid for stride-1 on
+            // the fastest dimension.  So only apply a solution if
+            // we're not doing even-length real, or if fastest dim
+            // stride is 1.
+            const bool evenLengthReal = (transformType == rocfft_transform_type_real_forward
+                                         || transformType == rocfft_transform_type_real_forward)
```

Comment:
```suggestion
            const bool evenLengthReal = (transformType == rocfft_transform_type_real_forward
                                         || transformType == rocfft_transform_type_real_backward)
```
I'm assuming that's what was intended (otherwise there is redundancy): if so, shouldn't the check below be on `execPlan.rootPlan->outputLenght[0] % 2` instead of `execPlan.rootPlan->length[0] % 2` for real backward cases?